vm_supervisor/network/hostnetwork.py

```diff
@@ -1,26 +1,144 @@
 import logging
+from ipaddress import IPv6Network
 from pathlib import Path
+from typing import Optional, Protocol
+
+from aleph_message.models import ItemHash
+
+from vm_supervisor.conf import IPv6AllocationPolicy
 
 from .firewall import initialize_nftables, setup_nftables_for_vm, teardown_nftables
 from .interfaces import TapInterface
 from .ipaddresses import IPv4NetworkWithInterfaces
+from .ndp_proxy import NdpProxy
 
 logger = logging.getLogger(__name__)
 
 
+def _read_proc_file_value(config_file: Path) -> int:
+    return int(config_file.read_text())
+
+
 def get_ipv4_forwarding_state() -> int:
-    """Reads the current ipv4 forwarding setting from the hosts, converts it to int and returns it"""
-    return int(Path("/proc/sys/net/ipv4/ip_forward").read_text())
+    """Reads the current IPv4 forwarding setting from the host, converts it to int and returns it"""
+    return _read_proc_file_value(Path("/proc/sys/net/ipv4/ip_forward"))
+
+
+def get_ipv6_forwarding_state() -> int:
+    """Reads the current IPv6 forwarding setting from the host, converts it to int and returns it"""
+    return _read_proc_file_value(Path("/proc/sys/net/ipv6/conf/all/forwarding"))
+
+
+class IPv6Allocator(Protocol):
+    def allocate_vm_ipv6_subnet(self, vm_id: int, vm_hash: ItemHash) -> IPv6Network:
+        ...
+
+
+class StaticIPv6Allocator(IPv6Allocator):
+    """
+    Static IPv6 allocator.
+    Computes IPv6 addresses based on the machine type and VM hash. The algorithm works
+    as follows:
+
+    | Component | CRN /64 range | VM type | Item hash prefix | Instance range |
+    |-----------|---------------|---------|------------------|----------------|
+    | Length    | 64 bits       | 16 bits | 44 bits          | 4 bits         |
+    """
+
+    def __init__(self, ipv6_range: IPv6Network, subnet_prefix: int):
+        if ipv6_range.prefixlen != 64:
+            raise ValueError(
+                "The static IP address allocation scheme requires a /64 subnet"
+            )
+        if subnet_prefix < 124:
+            raise ValueError("The IPv6 subnet prefix cannot be larger than /124.")
+
+        self.ipv6_range = ipv6_range
+        self.subnet_prefix = subnet_prefix
+
+    def allocate_vm_ipv6_subnet(self, vm_id: int, vm_hash: ItemHash) -> IPv6Network:
+        ipv6_elems = self.ipv6_range.exploded.split(":")[:4]
+        ipv6_elems += ["1"]  # Magic number for microVMs
```

Comment:
Magic number ?

Reply:
As described [here](https://community.aleph.im/t/vm-ipv6-routing/63/3) I used the xxxx:xxxx:xxxx:xxxx:1::/80 subnet for microVMs. We should add a parameter to the function to support the different VM types.

Reply:
We now use the VM type to determine this number, the discussion is moot.